test/e2e-helm/e2e_helm_suite_test.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e_helm_test
+
+import (
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo" //nolint:golint
+	. "github.com/onsi/gomega" //nolint:golint
+
+	testutils "github.com/operator-framework/operator-sdk/test/internal"
+	"github.com/operator-framework/operator-sdk/version"
+)
+
+// TestE2EHelm ensures the Helm projects built with the SDK tool by using its binary.
+func TestE2EHelm(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping Operator SDK E2E Helm Suite testing in short mode")
+	}
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "E2EHelm Suite")
+}
+
+var (
+	tc testutils.TestContext
+	// isPromethuesManagedBySuite is true when the suite tests is installing/uninstalling the Prometheus
+	isPromethuesManagedBySuite = true
+	// isOLMManagedBySuite is true when the suite tests is installing/uninstalling the OLM
+	isOLMManagedBySuite = true
+	// Kubectx staore the k8s context from where the tests are running
+	Kubectx string
+)
+
+// BeforeSuite run before any specs are run to perform the required actions for all e2e Helm tests.
+var _ = BeforeSuite(func(done Done) {
+	var err error
+
+	By("Creating a new test context")
+	tc, err = testutils.NewTestContext("GO111MODULE=on")
+	Expect(err).NotTo(HaveOccurred())
+	Expect(tc.Prepare()).To(Succeed())
+
+	// Check if is running in a Kind Cluster
+	By("Checking the cluster type")
+	Kubectx, err = tc.Kubectl.Command("config", "current-context")
+	Expect(err).Should(Succeed())
+
+	By("Checking if cluster has Prometheus")
+	output, err := tc.Kubectl.Command("api-resources")
+	Expect(err).NotTo(HaveOccurred())
+	if strings.Contains(output, "servicemonitors") {
+		isPromethuesManagedBySuite = false
+	}
+
+	if isPromethuesManagedBySuite {
+		By("Installing Prometheus")
+		Expect(tc.InstallPrometheusOperManager()).To(Succeed())
+
+		By("Ensuring provisioned Prometheus Manager Service")
+		Eventually(func() error {
+			_, err := tc.Kubectl.Get(
+				false,
+				"Service", "prometheus-operator")
+			return err
+		}, 3*time.Minute, time.Second).Should(Succeed())
+	}
+
+	By("Checking if cluster has OLM")
+	output, err = tc.Kubectl.Command("api-resources")
+	Expect(err).NotTo(HaveOccurred())
+	if strings.Contains(output, "clusterserviceversions") {
+		isOLMManagedBySuite = false
+	}
+
+	if isOLMManagedBySuite {
```

Comment:
Is this necessary for any tests right now? When testing bundle/package manifests generation + deployment it will be, but only for that test. In that case, this shouldn't be in `BeforeSuite` conditionally, it should be in `BeforeEach` in that specific test.

Reply:
To install the Prometheus and OLM takes too long and we can need both in more than one test. Because of this they are managed here where they can be installed once for all instead of many times in each required place. Note that if we add in `BeforeEach` each will be executed for each test.

Reply:
We don't need OLM for more than one test. In fact, we don't need OLM for any tests right now. Seems to be the same case for prometheus.

Reply:
Done the consolidation.